src/main/java/org/broadinstitute/hellbender/utils/pairhmm/VectorLoglessPairHMM.java

```diff
@@ -0,0 +1,287 @@
+        try {
+            Process process = new ProcessBuilder("/bin/sh", "-c", command).start();
+            if (process.waitFor() != 0) {
+                return false;
+            }
+            if (process.exitValue() != 0) {
+                return false;
+            }
+        }
+        catch (InterruptedException | IOException e) {
+            return false;
+        }
+        return true;
+    }
+    
+    private static Boolean isVectorLoglessPairHMMLibraryLoaded = false;
+
+    //The constructor is called only once inside PairHMMLikelihoodCalculationEngine
+    public VectorLoglessPairHMM() throws UserException.HardwareFeatureException {
+        super();
+
+        synchronized (isVectorLoglessPairHMMLibraryLoaded) {
+            // if AVX is not supported, throw an exception
+            if (!isAVXSupported()) {
+                throw new UserException.HardwareFeatureException("Machine does not support AVX PairHMM.");
+            }
+
+            // Load the library and initialize the FieldIDs
+            if (!isVectorLoglessPairHMMLibraryLoaded) {
+                try
+                {
+                    //Try loading from Java's library path first
+                    //Useful if someone builds his/her own library and wants to override the bundled
+                    //implementation without modifying the Java code
+                    System.loadLibrary("VectorLoglessPairHMM");
+                    logger.info("libVectorLoglessPairHMM found in JVM library path");
+                } catch (UnsatisfiedLinkError ule) {
+                    //Could not load from Java's library path - try unpacking from jar
+                    try
+                    {
+                        logger.debug("libVectorLoglessPairHMM not found in JVM library path - trying to unpack from GATK jar file");
+                        String extension = runningOnMac ? ".dylib" : ".so";
+                        loadLibraryFromJar("/lib/libVectorLoglessPairHMM" + extension);
+                        logger.info("libVectorLoglessPairHMM unpacked successfully from GATK jar file");
+                    } catch (IOException ioe) {
+                        //Throw the UnsatisfiedLinkError to make it clear to the user what failed
+                        throw ule;
+                    }
+                }
+                logger.info("Using vectorized implementation of PairHMM");
+                isVectorLoglessPairHMMLibraryLoaded = true;
+
+                //need to do this only once
+                jniInitializeClassFields(JNIReadDataHolderClass.class, JNIHaplotypeDataHolderClass.class);
+            }
+        }
+    }
+
+    //Hold the mapping between haplotype and index in the list of Haplotypes passed to initialize
+    //Use this mapping in computeLikelihoods to find the likelihood value corresponding to a given Haplotype
+    private HashMap<Haplotype, Integer> haplotypeToHaplotypeListIdxMap = new HashMap<>();
```

Comment:
We typically put all the field declarations together at the top of the class.  And any that can be final should be made final.